x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/TransportDeleteAsyncSearchAction.java

```diff
@@ -58,15 +67,40 @@ protected void doExecute(Task task, DeleteAsyncSearchAction.Request request, Act
         }
     }
 
-    private void cancelTaskAndDeleteResult(AsyncSearchId searchId, ActionListener<AcknowledgedResponse> listener) throws IOException {
+    void cancelTaskAndDeleteResult(AsyncSearchId searchId, ActionListener<AcknowledgedResponse> listener) throws IOException {
         AsyncSearchTask task = store.getTask(taskManager, searchId);
         if (task != null) {
-            task.cancelTask(() -> store.deleteResponse(searchId, false, listener));
+            //the task was found and gets cancelled. The response may or may not be found, but we will return 200 anyways.
+            task.cancelTask(() -> store.deleteResponse(searchId,
+                ActionListener.wrap(
+                    r -> listener.onResponse(new AcknowledgedResponse(true)),
+                    exc -> {
+                        //the index may not be there (no initial async search response stored yet?): we still want to return 200
+                        if (exc.getCause() instanceof IndexNotFoundException) {
```

Comment:
We shouldn't fail If the document is missing:
```suggestion
                        if (ExceptionsHelper.status(exc) == RestStatus.NOT_FOUND) {
```

Reply:
we don't, document missing does not come back as a failure. Yet, I agree on changing the condition as exc.getCause() instanceof IndexNotFoundException is not great

Reply:
++, thanks